File: M3 Intermediate DSA/src/L9_Arrays_Sliding_Window/Assignment3.java
package L9_Arrays_Sliding_Window;

public class Assignment3 {

  int printBoundary(int[][] A, int row, int col, int N, int val) {
    if (N == 1) {
      A[row][col] = val;
      return val;
    }


    for (int index = 1; index < N; index++) {
      A[row][col] = val;
      col++;
      val++;
    }

    for (int index = 1; index < N; index++) {
      A[row][col] = val;
      row++;
      val++;
    }

    for (int index = 1; index < N; index++) {
      A[row][col] = val;
      col--;
      val++;
    }

    for (int index = 1; index < N; index++) {
      A[row][col] = val;
      row--;
      val++;
    }

    return val;
  }

  public int[][] generateMatrix(int A) {
    int[][] ans = new int[A][A];
    int row = 0;
    int col = 0;
    int val = 1;

    while (A > 0) {
      val = printBoundary(ans, row, col, A, val);
      row++;
      col++;
      A -= 2;
    }

    return ans;
  }

}

File: M3_Intermediate_DSA/L7_Sliding_Window/assignment3.go
package main

func printClockwise(A *[][]int, i, j, N, val int) int {

	if N == 1 {
		(*A)[i][j] = val
		return val
	}

	for k := 1; k < N; k++ {
		(*A)[i][j] = val
		j++
		val++
	}

	for k := 1; k < N; k++ {
		(*A)[i][j] = val
		i++
		val++
	}

	for k := 1; k < N; k++ {
		(*A)[i][j] = val
		j--
		val++
	}

	for k := 1; k < N; k++ {
		(*A)[i][j] = val
		i--
		val++
	}

	return val
}

func GenerateMatrix(A int) [][]int {
	ans := make([][]int, A)

	for i := range ans {
		ans[i] = make([]int, A)
	}

	i := 0
	j := 0
	val := 1

	for A > 0 {
		val = printClockwise(&ans, i, j, A, val)
		i++
		j++
		A -= 2
	}

	return ans
}
